app/src/main/java/org/pragyan/dalal18/notifications/NotificationFragment.kt

```diff
@@ -144,13 +150,37 @@ class NotificationFragment : Fragment() {
         }
     }
 
+    private val refreshNotifications = object : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            customNotificationList.add(Notification(intent.getStringExtra(TEXT_KEY), intent.getStringExtra(CREATED_AT_KEY)))
+            customNotificationList.sortByDescending { it.createdAt }
+            notificationRecyclerAdapter.swapData(customNotificationList)
+        }
+    }
+
     override fun onPause() {
         super.onPause()
+        LocalBroadcastManager.getInstance(requireContext()).unregisterReceiver(refreshNotifications)
         preferences.edit().remove(Constants.LAST_NOTIFICATION_ID).apply()
     }
 
+    override fun onResume() {
+        super.onResume()
+        //IF IT IS ALIVE DUE TO SOME MISHAP IT IS UNREGISTERED
+        LocalBroadcastManager.getInstance(requireContext()).unregisterReceiver(refreshNotifications)
+        val intentFilter = IntentFilter(MainActivity.REFRESH_UNREAD_NOTIFICATIONS_COUNT)
+        LocalBroadcastManager.getInstance(requireContext()).registerReceiver(refreshNotifications, IntentFilter(intentFilter))
+    }
+
     override fun onDestroy() {
         super.onDestroy()
+        LocalBroadcastManager.getInstance(requireContext()).unregisterReceiver(refreshNotifications)
```

Comment:
In one of the projects I had run into this. Maybe, that's why you have to unregister in onResume()? Basically use onPause()?
https://stackoverflow.com/questions/17195641/fragment-lifecycle-when-ondestroy-and-ondestroyview-are-not-called

Reply:
I am using onPause too. It still doesn't seem to unregister at times.

Reply:
I've fixed it. The problem was that I was registering once in onView created too so this + onResume registered it twice.

Reply:
COol

Reply:
So remove this from onDestroy() now?

Reply:
It is still needed in onDestroy. The problem was it was getting Registered twice at the start.